tests/test_all_component_and_model_builder.py

```diff
@@ -0,0 +1,296 @@
+"""Test model builder using variance of parfile input"""
+
+from collections import defaultdict
+import pytest
+import io
+import astropy.units as u
+from pint.models.timing_model import (
+    PhaseComponent,
+    Component,
+    AllComponents,
+    ConflictAliasError,
+    UnknownBinaryModel,
+)
+from pint.models.model_builder import ModelBuilder, ComponentConflict
+from pint.models.parameter import floatParameter
+from pint.utils import split_prefixed_name, PrefixError
+
+
+class SimpleModel(PhaseComponent):
+    """Very simple test model component
+    """
+
+    register = True
+    category = "simple_test"
+
+    def __init__(self):
+        super(SimpleModel, self).__init__()
+        self.add_param(floatParameter(name="TESTPARAM", value=0.0, unit="s"))
+
+
+class SubsetModel(PhaseComponent):
+    """Test model component hosting the parameters which are a subset of spindown.
+    """
+
+    register = False  # This has to be false, otherwrise all test will fail.
+    category = "simple_test"
+
+    def __init__(self):
+        super(SubsetModel, self).__init__()
+        self.add_param(floatParameter(name="F0", value=0.0, unit="1/s"))
+        self.add_param(floatParameter(name="F1", value=0.0, unit="1/s^2"))
+
+
+@pytest.fixture
+def sub_set_model():
+    return SubsetModel()
+
+
+@pytest.fixture
+def simple_model():
+    return SimpleModel()
+
+
+@pytest.fixture
+def simple_model_overlap(simple_model):
+    simple_model.add_param(floatParameter(name="F0", aliases=[], value=0.0, unit="1/s"))
+    return simple_model
+
+
+@pytest.fixture
+def simple_model_alias_overlap():
+    simple_model = SimpleModel()
+    simple_model.add_param(
+        floatParameter(name="TESTPARAM2", aliases=["F0"], value=0.0, unit="1/s")
+    )
+    simple_model.add_param(
+        floatParameter(name="TESTPARAM3", aliases=["LAMBDA"], value=0.0, unit="deg")
+    )
+    return simple_model
+
+
+def test_model_builder_class():
+    """Test if AllComponents collected components information correctly
+    """
+    mb = AllComponents()
+    category = mb.category_component_map
+    assert len(mb.param_component_map["PX"]) == len(category["astrometry"])
```

Comment:
Can you test more than just the length? Should all the components, or all the names, be the same?